Move lost grapesjs documentation to docs, small improves

adiux opened the pull request:
Gitpod link: https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/43

<a href="https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/43"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

adiux committed:
move lost grapesjs documentation to docs, small improves

RCheesley commented:
Hey @adiux thanks for doing this! You have some Vale spelling / grammar / style errors that will need to be corrected (see the linter) - and there are some RST issues I can notice (headings are too long etc). If you are using VS Code you should get the highlights automatically to tell you where there are problems if you've got Vale installed.

Alternatively you could work in Gitpod: https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/43 which has everything configured and will flag up issues with the PR in the Problems tab (ignore the ones relating to other pages, we will fix those in a separate PR!)

Happy to walk through it if you get stuck with anything!

RCheesley committed:
Fix vale errors and warnings

RCheesley committed:
Fix incorrect link ref

RCheesley commented on docs/components/grapesjs.rst:
```diff
@@ -0,0 +1,74 @@
+Currently, Mautic uses the GrapesJS Builder for Emails and Landing pages. 
+
+Setup 
+*****
```
I think we need to be a bit more verbose here. Where exactly do they run `npm install`? Do they have to have Mautic installed first? 

We need to assume that people reading this are completely new to using the builder and may not know where to start.

RCheesley commented on docs/components/grapesjs.rst:
```diff
@@ -0,0 +1,74 @@
+
+.. vale on
+
+Use the template files provided. For example .eslintrc.temp
```
And do what with them? Can we be a bit more helpful here? Even if we point people out to documentation on external sites, it would be helpful to folks who might not know what to do with the template files.

RCheesley commented on docs/components/grapesjs.rst:
```diff
@@ -0,0 +1,74 @@
+
+Use the template files provided. For example .eslintrc.temp
+
+How to test standalone
```
What does it mean to test standalone? This isn't very clear. Perhaps we could explain why you would do this and what it would allow you to do?

RCheesley commented on docs/components/grapesjs.rst:
```diff
@@ -0,0 +1,74 @@
+
+.. code-block:: BASH
+
+    npm run start-helloWorld
```
As above I think we need to be clear - are we in Mautic's root? In the plugin folder?  What is the difference between the two commands and why might you use one rather than the other?

RCheesley commented on docs/components/grapesjs.rst:
```diff
@@ -0,0 +1,74 @@
+
+In order for ``start-mautic`` to work, a running DDEV container has to be present. 
+
+If you are using another development environment you need to update some paths in ``Demo/mautic/index.html``.
```
Let's be kind and give some examples here? You can use code blocks to include code samples.

RCheesley commented on docs/components/grapesjs.rst:
```diff
@@ -0,0 +1,74 @@
+
+.. code-block:: BASH
+
+    npm run build
```
As above, state where to run this.

RCheesley commented on docs/components/grapesjs.rst:
```diff
@@ -0,0 +1,74 @@
+
+All the JavaScript code lives in the Assets/library folder. This handles the UI of the builder.
+
+In addition to the code in this repository, there is also the :xref:`GrapesJS Preset`. 
```
```suggestion
In addition to the code in this `plugins/GrapesJsBuilderBundle` folder, there is also the :xref:`GrapesJS Preset`. 
```

We aren't referring to a separate repo at this time.

RCheesley commented on docs/components/grapesjs.rst:
```diff
@@ -0,0 +1,74 @@
+
+In addition to the code in this repository, there is also the :xref:`GrapesJS Preset`. 
+
+This repository handles the basic Mautic specific code. The general idea is that this preset repository is a base for various Mautic builder Plugins. 
```
```suggestion
This code in the `plugins/GrapesJsBuilderBundle` folder handles the basic Mautic specific code. The general idea is that the GrapesJS Preset repository is a base for various Mautic builder Plugins. 
```

RCheesley commented on docs/components/grapesjs.rst:
```diff
@@ -0,0 +1,74 @@
+
+This repository handles the basic Mautic specific code. The general idea is that this preset repository is a base for various Mautic builder Plugins. 
+
+E.g one where the Rich Text Editor is CKEditor, or where there is some very customer or industry specific features.
```
```suggestion
Examples might include if you wanted to have the Rich Text Editor is CKEditor, or where there is some very customer or industry specific features. These can be done with a custom preset.
```

RCheesley commented on docs/components/grapesjs.rst:
````diff
@@ -0,0 +1,74 @@
+- Opens the Block Manager at launch
+- Add Dynamic Content Block for HTML used in Mautic
+```
+
````
I think that we need to add here some instructions on how to roll your own custom preset. For example how to create a fork of the Mautic preset, and how to ship it with your own Mautic instance. I think that a few people do that already so hopefully we can get that documented.

I would also like to see some basic examples of:

How to create a new block
How to create a new section

These could link out to external documentation or be included here as our own docs.

RCheesley commented on docs/components/grapesjs.rst:
```diff
@@ -0,0 +1,74 @@
+PRs
+***
+
+Contributions to improve the GrapesJS Plugin happen in the main Mautic repository as :xref:`Github PRs`
```
I think we should also include a section on how to contribute and test PRs for the preset? That used to be in the readme I think.

RCheesley reviewed (changes requested):
Thanks for making the PR to bring across some of the details @adiux - much appreciated!

I have left some comments on the PR, I think we need to be a lot more detailed on the information we are provided so that we can support developers in learning how to extend and work with the builder and the preset - let me know if anything is unclear!

RCheesley committed:
Fix case-sensitive link

RCheesley commented:
@adiux @volha-pivavarchyk please could you take a look at the feedback/suggestions? It would be great to improve this documentation for our developers and get the PR merged but we are missing some key pieces of information I feel.

Commit:
REST API: port forms documentation

Commit:
Add menu item + small fixes

Commit:
REST API: Add Companies documentation

Commit:
Tiny fix

Commit:
Update A/B testing link

Looks like the old one no longer works, causing `make checklinks` to fail!

Commit:
Fix final linting errors

Commit:
Fix final linting errors

Commit:
Final linting fixes

RCheesley committed:
Fix broken link on dev env section

RCheesley committed:
Merge pull request #138 from mautic/update-ab-testing-link

Update A/B testing link

RCheesley committed:
Merge branch 'main' into port-forms-documentation

RCheesley committed:
Merge pull request #136 from mautic/port-forms-documentation

REST API: port forms documentation

RCheesley committed:
Merge branch 'main' into port-companies-documentation

RCheesley committed:
Merge pull request #137 from mautic/port-companies-documentation

REST API: Add Companies documentation

dennisameling committed:
Fix "anchor not found" error (#142)

dennisameling committed:
Add various REST API docs (emails, files, marketing messages, notes, pages, roles) (#139)

* Add various REST API docs

* Some final fixes

* Apply suggestions from code review

Co-authored-by: Ruth Cheesley <ruth@ruthcheesley.co.uk>

* Fix wrong docs on User Permissions and add small fix for Landing Pages

---------

Co-authored-by: Ruth Cheesley <ruth@ruthcheesley.co.uk>

Commit:
Another bunch of REST API docs

Commit:
Add links to menu and fix Companies heading

Commit:
More REST API docs!

Commit:
Getting Started and auth docs

RCheesley committed:
Apply suggestions from code review

Mostly vale fixes and small typos

RCheesley committed:
Merge pull request #143 from mautic/another-bunch-of-rest-api-docs

Add more REST API docs (stages, stats, tags, themes, tweets)

RCheesley committed:
Apply suggestions from code review

RCheesley committed:
Apply suggestions from code review

RCheesley committed:
Merge branch 'main' into almost-final-batch-of-rest-api-docs

RCheesley committed:
Merge pull request #144 from mautic/almost-final-batch-of-rest-api-docs

Add more REST API docs (Dashboard Widget Data, Dynamic Content, Point Actions, Point Triggers, Webhooks)

thmclellan committed:
REST API: Add list contacts by email example

Suggest we add an example of how to search contacts by email per https://forum.mautic.org/t/mautic-api-editing-contact-by-email/16997/6. Thanks

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

RCheesley committed:
Merge pull request #147 from mautic/all-contributors/add-thmclellan

docs: add thmclellan as a contributor for doc

RCheesley committed:
Merge branch 'main' into rest-api-contacts-by-email

thmclellan committed:
Update docs/rest_api/contacts.rst

Co-authored-by: Ruth Cheesley <ruth@ruthcheesley.co.uk>

thmclellan committed:
Update docs/rest_api/contacts.rst

Co-authored-by: Ruth Cheesley <ruth@ruthcheesley.co.uk>

Commit:
Finalize Getting Started and Auth docs

Commit:
Final Vale fixes

Commit:
Add "Run in Postman" button

RCheesley committed:
Merge pull request #146 from thmclellan/rest-api-contacts-by-email

REST API: Add list contacts by email example

RCheesley committed:
Merge branch 'main' into getting-started-and-auth-rest-api

Commit:
added 2 new params to email.rst

added info for *useOwnerAsMailer* and *utmTags* params to the following:
- Get email response
- Get email properties
- List email response
- Create email properties
- Edit email properties

escopecz committed:
Merge pull request #145 from mautic/getting-started-and-auth-rest-api

REST API: Getting Started and Auth docs

RCheesley committed:
Apply suggestions from code review

RCheesley committed:
Merge branch 'main' into added-2-new-params-to-email

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

RCheesley committed:
Merge pull request #152 from mautic/all-contributors/add-Entrepositive

docs: add Entrepositive as a contributor for doc

RCheesley committed:
Merge pull request #150 from Entrepositive/added-2-new-params-to-email

added 2 new params to email.rst

escopecz committed:
Completing Report API docs (#153)

* Completing Report API docs

* Fix Vale errors, warnings and suggestions

---------

Co-authored-by: Dennis Ameling <dennis.ameling@leap.ac>

dennisameling committed:
GitHub Actions: update Actions versions (#154)

RCheesley committed:
Password complexity

RCheesley committed:
Password complexity for ddev page

RCheesley committed:
Merge pull request #156 from mautic/password-complexity

Update password complexity

RCheesley commented:
Hey folks @adiux @volha-pivavarchyk could we please get this info into the docs? It will need rebasing to 5.x and the feedback addressing. What do you think?

RCheesley committed:
Merge branch 'main' into init-grapesjs

RCheesley commented:
Closing due to lack of response. If we still need these changes, please cherry-pick to a new PR.
